### ISSUE TITLE
I tried to add a new training parameter in yolov8. The code has been modified, but ultralytics shows that there is no such parameter, but I have modified the actual code. What do I need to do? Let the training happen using my own code.

### ISSUE THREAD

**Issue by hu874** [ultralytics/ultralytics#18143]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I tried to add a new training parameter in yolov8. The code has been modified, but ultralytics shows that there is no such parameter, but I have modified the actual code. What do I need to do? Let the training happen using my own code.

### Environment

New https://pypi.org/project/ultralytics/8.3.48 available 😃 Update with 'pip install -U ultralytics'
Ultralytics 8.3.40 🚀 Python-3.10.15 torch-2.5.0 CUDA:2 (NVIDIA GeForce RTX 4090, 24210MiB)
                                                 CUDA:3 (NVIDIA GeForce RTX 4090, 24210MiB)
                                                 CUDA:4 (NVIDIA GeForce RTX 4090, 24210MiB)
                                                 CUDA:5 (NVIDIA GeForce RTX 4090, 24210MiB)
engine/trainer: task=detect, mode=train, model=yolov8m-p2.yaml, data=/opt/data/private/jyh/code/huqian/ultralytics-yolo_multilabel/data/data.yaml, epochs=300, time=None, patience=100, batch=160, imgsz=640, save=True, save_period=-1, cache=False, device=2,3,4,5, workers=13, project=/opt/data/private/jyh/model/s31_trafficlightv8_multlabels_20241008_cls46_beat, name=yolov8m-p2-multlabels-trafficlight_26_1210, exist_ok=False, pretrained=True, optimizer=AdamW, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, multilabel=True, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.001, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.03014, warmup_momentum=0.91406, warmup_bias_lr=0.1, box=4.82505, cls=1.45405, dfl=0.9, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.00855, hsv_s=0.34573, hsv_v=0.25633, degrees=0.0, translate=0.0, scale=0.0, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=0.64575, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=/opt/data/private/jyh/model/s31_trafficlightv8_multlabels_20241008_cls46_beat/yolov8m-p2-multlabels-trafficlight_26_1210
Overriding model.yaml nc=80 with nc=26

### Minimal Reproducible Example

![image](https://github.com/user-attachments/assets/32072fa8-a20e-4db8-be34-1b25dc8777c6)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @hu874, thank you for your interest in Ultralytics 🚀! We appreciate you taking the time to report this issue.

We recommend reviewing the [Ultralytics Docs](https://docs.ultralytics.com) to ensure you are familiar with how to customize training parameters in YOLO models. Specifically, the [Python Usage Guide](https://docs.ultralytics.com/usage/python/) could offer insights into parameter modifications.

If this is a 🐛 Bug Report, we kindly request that you provide a clear [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us trace and debug the issue. Including the exact steps to reproduce the problem, along with the relevant code snippets or configuration changes, will greatly assist us in diagnosing your issue.

Additionally, please ensure you've upgraded to the latest `ultralytics` package to confirm the issue is not already resolved. You can update your installation by running:

```bash
pip install -U ultralytics
```

### Environments

YOLO models may be run in any of the following up-to-date verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Ensure that any changes you've made to the codebase match the expected behavior of the training pipeline. It's also a good idea to cross-verify that the parameters you added are properly registered within the model or training pipeline.

💡 For collaborative discussions or to seek quick assistance, you can join our Ultralytics community:
- Real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions on [Discourse](https://community.ultralytics.com)
- Engaging conversations on our [Subreddit](https://reddit.com/r/Ultralytics)

An Ultralytics engineer will review your ticket soon and provide further support as needed—stay tuned! 😊

**Comment by Y-T-G** [comment]
You should install your modified clone by running
`pip install -e .` (dot at the end) inside the ultralytics folder.
